src/python/gem5/resources/resource.py

```diff
@@ -884,6 +937,144 @@ def __init__(
         self._func = function
         self._params = parameters
 
+        self.get_all_resources(
+            resources=resources,
+            to_path=local_path,
+            resource_directory=resource_directory,
+            download_md5_mismatch=download_md5_mismatch,
+            clients=clients,
+            gem5_version=gem5_version,
+            quiet=quiet,
+        )
+
+    def get_all_resources(
+        self,
+        resources: Dict[str, Dict[str, str]],
+        to_path: str,
+        resource_directory: str,
+        download_md5_mismatch: bool,
+        clients: List[str],
+        gem5_version: str,
+        quiet: bool,
+    ) -> None:
+        """
+        This function gets all the resources for the workload.
+
+        :param resources: A dictionary of resources to get. The keys are the
+                            parameter names and the values are the resource
+                            id and version.
```

Comment:
Ok, this is better!

What is a "parameter name" and why is the resource id/version a dictionary? Why is the id a key and the version a value?

An example would go a long way here.

Reply:
This is an exact copy of the field in JSON. In JSON we have a resoruces field which looks like this:
```
resources: {
    "binary":{
        "id":"id",
        "resource_version":"1.0.0"
    },
    "readin_file":{
        "id":"id2",
        "resource_version":"1.0.0"
    },
}
```

so the key is the name of the paramter that is used in the set workload functions and the value is the resource that we have to pass it.

I am going to refactor this so that I dont have to pass this many variables around into the worklaod and directly just give it the objects.